lib/publiccloud/provider.pm

```diff
@@ -248,6 +249,42 @@ sub destroy_instance {
     $self->ipa(cleanup => 1, running_instance_id => $instance->instance_id);
 }
 
+sub _get_config {
+    my ($self) = @_;
+
+    unless ($self->{rest_config}) {
+        my $config_file = $ENV{OPENQA_CONFIG} || '/etc/openqa';
+        $config_file .= '/publiccloud.ini';
+        my $cfg = Config::IniFiles->new(-file => $config_file)
+          or die("Failed to load $config_file");
+        $self->{rest_config} = {
+            url => $cfg->val('global', 'url'),
```

Comment:
and if file exists but there is no `url` variable ? will we fail ?

Reply:
I would say yes, as it is mandatory to have one.